mlflow/server/handlers.py

```diff
@@ -1103,6 +1104,140 @@ def _default_history_bulk_impl():
     }
 
 
+def _get_sampled_steps_from_steps(start_step, end_step, max_results):
+    # Implementation of https://src.dev.databricks.com/databricks/universe@92cf18fbbd640800be1b2c80b698cc78e91020e4/-/blob/mlflow/src/main/scala/com/databricks/mlflow/MlflowBackend.scala?L1995:15&popover=pinned#tab=def
+    num_steps = end_step - start_step + 1
+    interval = num_steps / max_results
+    steps = []
+    intervals = [-1]
+    for i in range(start_step, end_step):
+        idx = (i - start_step) // interval
+        if idx != intervals[-1]:
+            intervals.append(idx)
+            steps.append(i)
+    return steps
```

Comment:
/ai give me some input and output example pairs for this function.

Reply:
/ai thanks, I want python code verify these examples

Reply:
Those samples are not correct 🤣

Reply:
@serena-ruan Can you add a couple examples? Examples are worth 1M words.

Reply:
Yes, added!

Reply:
```
def _get_sampled_steps_from_steps(start_step, end_step, max_results):
    interval = (end_step - start_step + 1) / max_results
    return [int(start_step + i * interval) for i in range(max_results)]
```

Is this equivalent to

Reply:
It's not, for example start_step=0, end_step=100, max_results=5, interval is 20.2, then the result should be [0, 21, 41, 61, 81] while this function is [0, 20, 40, 60, 80]. I think we don't need to simplify the implementation of this function as keeping the logic same as scala code could make sure it doesn't break :)